### DISCUSSION TITLE
dissolution

### DISCUSSION THREAD

**Discussion by mymyrere** [OpenDrift/opendrift#853] [Q&A]
Hi,
I was wondering if anyone has used the NOAA dissolution physics [(NOAA dissolution)](https://github.com/NOAA-ORR-ERD/PyGnome/blob/625c6911305a964db25590ed639f73021af90d8f/py_gnome/gnome/weatherers/dissolution.py) inside OpenDrift.

Is there a reason why this wasn't included in OpenDrift? 
Looks like this could be inserted into Opendrift? I wanted to give this a go but wanted to check if someone has already done it.

Thanks.

**Comment by knutfrode** [comment]
Hi,

@kkotzak has implemented dissolution in his fork: https://github.com/CIGOM-Modelacion/cicoil-module/blob/2448d739487ce85461931cfc4fb78730bf698c9a/opendrift/models/ciceseoil.py

So perhaps this could be copied/merged into the main OpenDrift, with @kkotzak 's permission?

Alternatively it should also be ok to "borrow" the NOAA implementation, given proper referencing. There are already several references to NOAA in OpenOil.

**Reply by mymyrere** [reply]
Thank you for showing me this fork.
I will try @KKotzak method, would be great to add this option in Opendrift.

**Reply by KKotzak** [reply]
Hi Remy,

Hi Remy, the dissolution and the other weathering processes in the CIGOM fork are based on a multi-component approach and they are made up of few files that extend (without modifying the original opendrift files) OpenDrift version 1.5.6. I am working on making them compatible with the latest OpenDrift version 1.8.2 with Adios oil database so they can be merged into the official repository. This task was supposed to be completed during the new year holidays but it will be completed soon. Then the whole set of multi-component based weathering processes will be accessed through a new weathering option e.g. "cicese".

**Reply by mymyrere** [reply]
Nice work on this version.
Is there any reference/ publications on this method?

**Comment by mymyrere** [comment]
I have tried @KKotzak  but found major difference in the evaporation process.
So I have updated OpenOil and added the dissolution in it.
I haven't tested it extensively but did a few test and results seems to be OK.

Happy for the code to be in OpenDrift.
Need to rename the code attached to .py

[noaa_oil_weathering.txt](https://github.com/OpenDrift/opendrift/files/8036599/noaa_oil_weathering.txt)
[openoil .txt](https://github.com/OpenDrift/opendrift/files/8036600/openoil.txt)

Remy
